### ISSUE TITLE
Support ignoring test files and other directories for `gosec` linter

### ISSUE THREAD

**Issue by HydroH** [golangci/golangci-lint#5064]
### Welcome

- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Your feature request related to a problem? Please describe

`gosec` by default ignores test files, and supports ignoring additional folders with the `-exclude-dir` command line argument. This feature is not provided in `golangci-lint`.

### Describe the solution you'd like

Add `exclude-tests` (boolean) and `exclude-dir` (array) options to `gosec` config with respective support.

### Describe alternatives you've considered

The previous solution is straightfoward, I could not think of other alternatives.

### Supporter

- [ ] I am a sponsor through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Hello,

you can already exclude test files for a specific linter:

```yml
issues:
  exclude-rules:
    - path: (.+)_test.go
      linters:
        - gosec
```

https://golangci-lint.run/usage/false-positives/
